applications/FluidDynamicsApplication/python_scripts/apply_outlet_process.py

```diff
@@ -68,26 +80,63 @@ def ExecuteInitializeSolutionStep(self):
         self.aux_pressure_process.ExecuteInitializeSolutionStep()
         self.aux_external_pressure_process.ExecuteInitializeSolutionStep()
 
-        # Compute the current process average velocity
-        outlet_avg_vel_norm = 0.0
-        for node in self.outlet_model_part.GetCommunicator().LocalMesh().Nodes:
-            vnode = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY)
-            outlet_avg_vel_norm += math.sqrt(vnode[0]**2+vnode[1]**2+vnode[2]**2)
-        outlet_avg_vel_norm /= len(self.outlet_model_part.GetCommunicator().LocalMesh().Nodes)
-
-        # Perform MPI communication
-        self.outlet_model_part.GetCommunicator().SumAll(outlet_avg_vel_norm)                 # Accumulate the average velocity in each processor
-        outlet_avg_vel_norm /= self.outlet_model_part.GetCommunicator().TotalProcesses()     # Compute the average between processors
+        # If considered, add the hydrostatic component to the outlet pressure
+        if (self.hydrostatic_outlet):
+            self._AddOutletHydrostaticComponent()
 
-        # Store the average velocity in the ProcessInfo to be used in the outlet inflow prevention condition
-        min_outlet_avg_vel_norm = 1.0
-        if (outlet_avg_vel_norm >= min_outlet_avg_vel_norm):
-            self.outlet_model_part.GetRootModelPart().ProcessInfo[KratosFluid.CHARACTERISTIC_VELOCITY] = outlet_avg_vel_norm
-        else:
-            self.outlet_model_part.GetRootModelPart().ProcessInfo[KratosFluid.CHARACTERISTIC_VELOCITY] = min_outlet_avg_vel_norm
+        # Compute the outlet average velocity
+        self._ComputeOutletCharacteristicVelocity()
 
 
     def ExecuteFinalizeSolutionStep(self):
         # Call the base process ExecuteFinalizeSolutionStep()
         self.aux_pressure_process.ExecuteFinalizeSolutionStep()
         self.aux_external_pressure_process.ExecuteFinalizeSolutionStep()
+
+
+    # Private methods section
+    def _AddOutletHydrostaticComponent(self):
+        # Get the body force value
+        for node in self.outlet_model_part.Nodes:
+            body_force = node.GetSolutionStepValue(KratosMultiphysics.BODY_FORCE, 0)
+            break
+
+        # Compute the body force unit normal vector
+        body_force_norm = math.sqrt(math.pow(body_force[0],2)+math.pow(body_force[1],2)+math.pow(body_force[2],2))  # Body force norm
```

Comment:
math.pow(body_force[0],2) can be replaced by body_force[0]*body_force[0]  (maybe it wil be faster)